### DISCUSSION TITLE
Why is effective_creep_strain negative?

### DISCUSSION THREAD

**Discussion by zong-yanyun** [idaholab/moose#24062] [Q&A General] [answered]
In my tensor_mechanics module, creep_ rate is not a negative number in computeResidual, but effective_ creep_ Strain decreases. Could you tell me if this is reasonable in moose? And why? I personally verified that the absolute value should be correct, perhaps the symbol is incorrect?

**Comment by jiangwen84** [comment]
I am not sure about the answer off the top of my head, but I think it might be possible. Which creep model are you using?

**Reply by zong-yanyun** [reply]
The creep model was developed by ourselves. And I verified that my creep rate is not negative.

**Reply by zong-yanyun** [reply]
I have added some detailed descriptions. Could you please take a look again.

**Comment by zong-yanyun** [comment]
![image](https://user-images.githubusercontent.com/62926015/232508941-cb0fa473-cd89-4e27-a1f9-48100a3da967.png)
This is the effective creep strain result based on our own creep model. In our creep model, the effective creep strain rate is computed based on an empirical correlation, which is guaranteed to be positive. In our understanding, 
![image](https://user-images.githubusercontent.com/62926015/232509156-ea5894c9-5790-47e7-bf4e-989fb8ac17ef.png)
So if the creep rate is positive, the effective creep strain increment must be positive. But the effective strain decreases in our result ...
Is our understanding wrong? Or there are some BUGs in MOOSE?

**Reply by hugary1995** [reply]
You need to do proper code verification. You can integrate your in-house creep model using your favorite script, say python or matlab, with a constant stress load. Then compare the integrated effective creep strain with a single element test in moose. That's the usual way of benchmarking proprietary code.

**Comment by zong-yanyun** [comment]
I conducted the test according to what you said, and the results were no problem.
![image](https://user-images.githubusercontent.com/62926015/232734677-3e5f27bc-f255-413e-b9a5-ffd4a4fce2eb.png)
I really can't find the reason for the decline of effective creep strain. There are some BUGs in MOOSE?

**Reply by hugary1995** [reply, marked as answer]
It's very difficult for us to guess what you are doing exactly. At this point, it might be easier to set up the non-proprietary part of your model in a git repository and share with us. With a minimal non-working example, it will be much easier to debug, if there's any.
